app/models/foreman_kubevirt/kubevirt.rb

```diff
@@ -181,6 +181,7 @@ def create_vm(args = {})
                           :interfaces  => interfaces)
         client.servers.get(options[:name])
       rescue Fog::Kubevirt::Errors::ClientError => e
+        destroy_vm(vm.to_h[:metadata][:uid])
```

Comment:
if an exception occurred during client.vms.create(), the vm will be nil or even unassigned therefore it cannot be used in this context or at least check for its presence before accessing it.

Second, the parameter for destroy_vm should be the vm name which we already have in **options[:name]**, therefore the above could be written as: 

  `destroy_vm(options[:name])`

Reply:
destory_vm is used when you remove the vm as well, I can't change it.

Reply:
you just need to change the argument you're passing to this method, instead of: vm.to_h[:metadata][:uid] you should pass options[:name].